packages/sns/lib/utils/snsInitter.ts

```diff
@@ -150,26 +194,30 @@ export async function deleteSns(
     )
   }
 
-  if (!creationConfig.topic.Name) {
-    throw new Error('topic.Name must be set for automatic deletion')
-  }
+  if (creationConfig.topic) {
+    if (!creationConfig.topic.Name) {
+      throw new Error('topic.Name must be set for automatic deletion')
+    }
 
-  await deleteTopic(snsClient, creationConfig.topic.Name)
+    await deleteTopic(snsClient, creationConfig.topic.Name)
+  }
 }
 
 export async function initSns(
   snsClient: SNSClient,
-  locatorConfig?: SNSQueueLocatorType,
+  locatorConfig?: SNSTopicLocatorType,
   creationConfig?: SNSCreationConfig,
 ) {
   if (locatorConfig) {
-    const checkResult = await getTopicAttributes(snsClient, locatorConfig.topicArn)
+    const topicArn =
+      locatorConfig.topicArn ?? (await getTopicArnByName(snsClient, locatorConfig.topicName))
```

Comment:
Is it needed? It looks like new API not related to support both creation/locate configs.

Reply:
it is needed to support both having a predefined arn and a lookup by topic name, which new logic relies upon

Reply:
OK, so precedence will always be:
- use ARN if present
- then resolve by name (async lookup)